File: user-guide-models/multiple_regression_qr/multiple_regression_qr.py
import pystan
import numpy as np

N = 100
K = 4
alpha = -3
beta = np.array([1,2,3,4], dtype="float")
sigma = 5

x = np.random.uniform(size = N * K)
x = x.reshape((N, K))
y = np.random.normal(size = N, loc=alpha + x.dot(beta), scale = sigma)

stan_data = {'N': N, 'K': K, 'x': x, 'y': y}

model = pystan.StanModel(file='multiple_regression_qr.stan')

fit = model.sampling(data=stan_data)
print(fit)

File: user-guide-models/multiple_regression_qr/multiple_regression_qr.R
library("rstan")

N <- 100
K <- 4
x <- array(rnorm(N*K), c(N, K))
alpha <- -3
beta <- 1:K
sigma <- 5
y <- c(alpha + x %*% beta + rnorm(N, 0, sigma))  # Needed to do c() to convert y from a 1-column matrix into a vector
stan_data <- list(N=N, K=K, x=x, y=y)

model <- stan_model("multiple_regression_qr.stan")
fit <- sampling(model, data=stan_data)
print(fit)

cat(
    "alpha =", alpha, "\n",
    "beta =", beta, "\n",
    "sigma =", sigma, "\n"
)

